### PR TITLE
Add prettier as CSS/JS formatter

### PR DESCRIPTION
**Pull request by timoludwig**
### Short description

This adds [prettier](https://prettier.io/) as formatter for our static CSS & JS files

### Proposed changes

- Add prettier as dependency
- Add prettier dev tool
- Apply prettier code style to all existing files
- Add pre-commit hook for prettier
- Add prettier check to CircleCI workflow

### Side effects
- Increased CircleCI credit usage
- Merge conflicts with most currently open PRs

### Resolved issues

Part of: #749

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by charludo — APPROVED** [review]
Great PR, together with `feature/djlint` - won't have to disable those formatters anymore when working on the project 😆

File: dev-tools/prettier.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# This script can be used to format the static files with prettier
+
+# Import utility functions
+# shellcheck source=./dev-tools/_functions.sh
+source "$(dirname "${BASH_SOURCE[0]}")/_functions.sh"
+
+# Run prettier
+echo "Starting code formatting with prettier..." | print_info
+npx prettier --print-width 120 --write integreat_cms/static/src
```
[the remark this answers is not in the document]
Comment (timoludwig) [reply dev-tools/prettier.sh]: Yes, good idea! Interestingly, this increased the diff quite a lot, so it seems that we had mainly used 2 tabs as indentation before :thinking: 
But our Python files also use 4 spaces, so I guess it makes sense to use one consistent tab width for the entire project.

**Review by david-venhoff — APPROVED** [review]
Looks great! Maybe it would make sense to combine black and prettier (and eventually djlint) into a single formatting script?

**Comment by timoludwig** [comment]
> Looks great! Maybe it would make sense to combine black and prettier (and eventually djlint) into a single formatting script?

You mean like I did [here](https://github.com/digitalfabrik/integreat-cms/pull/1765/files#diff-74c9d308747f9e845de998fad5597022c7a586982e71ddf91d314bf1fef13977)? :slightly_smiling_face:

**Comment by david-venhoff** [comment]
Oh, right... :sweat_smile:

**Pull request merged**